src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public Ball getRandomBall() {
```

Comment:
This method can be inlined to 
public Ball getRandomBall() {
        return new Ball(ColorSupplier.getRandomColor(), new Random().nextInt(NUMBER_LIMIT));
    }
    
Using parametrised constructor over setters is advised (and even obligatory, when fields of class are final).